R/plot.data_table_threads_benchmark.R

```diff
@@ -38,12 +38,18 @@ plot.data_table_threads_benchmark <- function(x, ...)
 
   maxSpeedup <- x[, .(threadCount = threadCount[which.max(speedup)], speedup = max(speedup), type = "Ideal"), by = expr]
 
+  subOptimalSpeedupData <- data.table(
+    threadCount = seq(1, systemThreadCount, length.out = systemThreadCount),
+    speedup = seq(1, systemThreadCount / 2, length.out = systemThreadCount),
+    type = "Sub-optimal"
+  )
+  
   closestPoints <- x[, {
-    recommendedSpeedupSubset <- speedupData[expr == .BY$expr & type == "Recommended"]
-    merged <- .SD[recommendedSpeedupSubset, on = .(threadCount), nomatch = 0L]
-    .SD[which.max(speedup - merged$speedup)]
-  }, by = expr][, type := "Recommended"]
-
+    suboptimalSubset <- subOptimalSpeedupData[threadCount %in% .SD$threadCount]
+    .SD[which.max(speedup - suboptimalSubset$speedup)]
```

Comment:
please change suboptimal to recommended to avoid confusion

I don't think this logic is correct. why are you subtracting two speedup values before taking which.max?

the logic should be: first consider rows which are above the efficiency reference line (0.5 default), then take max speedup.

Reply:
I tried to address these in #20, so can you please review that?